File: leetcode/implementation/922. Sort Array By Parity II/922. Sort Array By Parity II.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        vector <int> ans;
        stack <int> odd, even;
        for(auto n : nums) {
            if(n%2) odd.push(n);
            else even.push(n);
        }
        for(int i = 0; i < nums.size() / 2; i++) {
            ans.push_back(even.top());
            even.pop();
            ans.push_back(odd.top());
            odd.pop();
        }
        return ans;
    }
};

File: leetcode/implementation/922. Sort Array By Parity II/922. Sort Array By Parity II.rs
impl Solution {
    pub fn sort_array_by_parity_ii(nums: Vec<i32>) -> Vec<i32> {
        let mut ans: Vec<i32> = Vec::new();
        let mut odd = Vec::new();
        let mut even = Vec::new();
        for n in nums {
            if n % 2 == 1 {
                odd.push(n);
            } else {
                even.push(n);
            }
        }
         while !odd.is_empty() || !even.is_empty() {
            if let Some(e) = even.pop() {
                ans.push(e);
            }
            if let Some(o) = odd.pop() {
                ans.push(o);
            }
        }
        ans
    }
}